src/client_side.h

```diff
@@ -321,8 +322,8 @@ class ConnStateData : public Server, public HttpControlMsgSink, private Independ
     /// tunneling them to the server later (on_unsupported_protocol)
     bool shouldPreserveClientData() const;
 
-    // TODO: move to the protected section when removing clientTunnelOnError()
-    bool tunnelOnError(const HttpRequestMethod &, const err_type);
+    // TODO: move to the protected section
+    bool tunnelOnError(const err_type);
```

Comment:
Let's not remove clientTunnelOnError() in this PR. It has a possibly useful assertion and requires more removal-related changes than you are willing to do (AFAICT).

Reply:
It's already done, so reverting/extracting is extra work. If you feel strongly I can do it, but it's okay for me. I would rather move forward, for instance, if the context assert is actually useful, I'd spend the time adding that back to the tunnelOnError method

Reply:
Please revert. I do not have the time to (guide you to) do this refactoring right, and doing it right is likely to result in non-trivial out-of-scope changes and/or analysis.

I regret that reverting this change creates extra work for you, but it is the risk you took when you decided to refactor this code in a project that does not require such refactoring. IMHO, finding the right project scope and sticking to it is one of the hardest and most important things we have to do. Let's be extra careful about this to minimize conflicts and reversals. FWIW, as a reviewer, I strive not to require any out-of-scope changes, so once the scope is correctly identified, taking a strict, minimalistic approach should not cause such problems during my reviews.

Reply:
Reverting. I'll create a new PR with the refactoring, we can discuss there about a proper refactor design